plottr/plot/mpl/autoplot.py

```diff
@@ -0,0 +1,466 @@
+                labels: Optional[List[str]] = None,
+                plotDataType: PlotDataType = PlotDataType.unknown,
+                **plotOptions: Any) -> int:
+
+        if self.plotType == PlotType.multitraces and join is None:
+            join = self.previousPlotId()
+        return super().addData(*data, join=join, labels=labels,
+                               plotDataType=plotDataType, **plotOptions)
+
+    def makeSubPlots(self, nSubPlots: int) -> List[Axes]:
+        """Create subplots (`Axes`). They are arranged on a grid that's close to square.
+
+        :param nSubPlots: number of subplots to make
+        :return: list of matplotlib axes.
+        """
+        if nSubPlots > 0:
+            nrows = int(nSubPlots ** .5 + .5)
+            ncols = int(np.ceil(nSubPlots / nrows))
+            gs = GridSpec(nrows, ncols, self.fig)
+            axes = [self.fig.add_subplot(gs[i]) for i in range(nSubPlots)]
+        else:
+            axes = []
+        return axes
+
+    def formatSubPlot(self, subPlotId: int) -> None:
+        """Format a subplot. Parses the plot items that go into that subplot,
+        and attaches axis labels and legend handles.
+
+        :param subPlotId: ID of the subplot.
+        """
+        labels = self.subPlotLabels(subPlotId)
+        axes = self.subPlots[subPlotId].axes
+
+        if isinstance(axes, list) and len(axes) > 0:
+            if len(labels) > 0 and len(set(labels[0])) == 1:
+                axes[0].set_xlabel(labels[0][0])
+            if len(labels) > 1 and len(set(labels[1])) == 1:
+                axes[0].set_ylabel(labels[1][0])
+
+        if isinstance(axes, list) and len(labels) == 2 and len(set(labels[1])) > 1:
+            axes[0].legend(loc='upper right', fontsize='small')
+
+        if isinstance(axes, list) and len(axes) > 1:
+            if len(labels) > 2 and len(set(labels[2])) == 1:
+                axes[1].set_ylabel(labels[2][0])
+        return None
+
+    def plot(self, plotItem: PlotItem) -> Optional[Union[Artist, List[Artist]]]:
+        """Plots data in a PlotItem.
+
+        :param plotItem: the item to plot.
+        :return: matplotlib Artist(s), or ``None`` if nothing was plotted.
+        """
+        if self.plotType in [PlotType.singletraces, PlotType.multitraces]:
+            return self.plotLine(plotItem)
+        elif self.plotType in [PlotType.image, PlotType.scatter2d, PlotType.colormesh]:
+            return self.plotImage(plotItem)
+        else:
+            return None
+
+    # methods specific to this class
+    def plotLine(self, plotItem: PlotItem) -> Optional[List[Line2D]]:
+        axes = self.subPlots[plotItem.subPlot].axes
+        assert isinstance(axes, list) and len(axes) > 0
+        assert len(plotItem.data) == 2
+        lbl = plotItem.labels[-1] if isinstance(plotItem.labels, list) and len(plotItem.labels) > 0 else ''
+        x, y = plotItem.data
+        return axes[0].plot(x, y.real, label=lbl, **plotItem.plotOptions)
```

Comment:
what is `.real` really needed for? isn't the data already split into real/imag/mag/phase parts? same in the plotImage method
```suggestion
        return axes[0].plot(x, y, label=lbl, **plotItem.plotOptions)
```

Reply:
good point -- this looks wrong (although there was still a -- hidden -- possibility that data might be complex typed).
i fixed it in the correct spot (apparently numpy.ma.abs can produce complex valued data).